docs/sources/flow/reference/components/discovery.kubelet.md

```diff
@@ -0,0 +1,147 @@
+---
+title: discovery.kubelet
+---
+
+# discovery.kubelet
+
+`discovery.kubelet` discovers Kubernetes pods running on the specified Kubelet
+and exposes them as scrape targets.
+
+## Usage
+
+```river
+discovery.kubelet "LABEL" {
+  kubelet_url = "KUBELET_URL"
+  bearer_token_file = "TOKEN_FILE"
+}
+```
+
+## Requirements
+
+* The Kubelet must be reachable from the `grafana-agent` pod network.
+* Follow the [Kubelet authorization](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/#kubelet-authorization)
+  documentation to configure authentication to the Kubelet API.
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`kubelet_url` | `string` | URL of Kubernetes API server. | | yes
+`bearer_token` | `secret` | Bearer token to authenticate with. | | no
+`bearer_token_file` | `string` | File containing a bearer token to authenticate with. | | no
+`refresh_interval` | `duration` | How often the Kubelet should be polled for scrape targets | `5s` | no
+`namespaces` | `list(string)` | A list of namespaces to extract target pods from | | no
+
+One of the following authentication methods must be provided if kubelet authentication is enabled
+ - [`bearer_token` argument](#arguments).
+ - [`bearer_token_file` argument](#arguments).
+ - [`authorization` block][authorization].
+
+### namespaces
+
+The `namespaces` list limits the namespaces to discover resources in. If
+omitted, all namespaces are searched.
```

Comment:
It's a little unusual for arguments to have their own header; normally they would just be described in the same header as the arguments or the block they're in:

```suggestion
The `namespaces` argument limits the namespaces to discover resources in. If
omitted, all namespaces are searched.
```